pkg/tfvars/openstack/rhcos_image.go

```diff
@@ -110,6 +73,52 @@ func uploadBaseImage(cloud string, localFilePath string, imageName string, clust
 	return nil
 }
 
+func importImage(imageName, imageID string, conn *gophercloud.ServiceClient, f *os.File) error {
+	logrus.Debugf("Using Image Import API to upload RHCOS to the image %q with ID %q", imageName, imageID)
+	stageRes := imagedata.Stage(conn, imageID, f)
+	if stageRes.Err != nil {
+		logrus.Debugf("Image uploading failed")
+		return stageRes.Err
+	}
+	logrus.Debugf("The data was uploaded")
+
+	logrus.Debugf("Begin image import for the image %q with ID %q", imageName, imageID)
+	co := imageimport.CreateOpts{
+		Name: imageimport.GlanceDirectMethod,
+	}
+	importRes := imageimport.Create(conn, imageID, co)
+	if importRes.Err != nil {
+		return importRes.Err
+	}
+	logrus.Debugf("Image import started")
+
+	// Image import is an asynchronous operation, so we have to wait until the image becomes "active"
+	const numRetries = 5000
+	const timeSleepSeconds = 15
```

Comment:
It is never a good sign when you have to define a "sleep" variable. 
Any other way to do this without it.